diff -ur dbus-1.5.12.orig/bus/selinux.c dbus-1.5.12/bus/selinux.c
--- dbus-1.5.12.orig/bus/selinux.c	2012-02-08 20:36:10.000000000 +0100
+++ dbus-1.5.12/bus/selinux.c	2012-05-27 23:02:39.487187070 +0200
@@ -1053,7 +1053,8 @@
       int rc;
 
       capng_clear (CAPNG_SELECT_BOTH);
-      capng_update (CAPNG_ADD, CAPNG_EFFECTIVE | CAPNG_PERMITTED,
+      // | causes rc to get a value of -9, & works (rc==0)
+      capng_update (CAPNG_ADD, CAPNG_EFFECTIVE & CAPNG_PERMITTED,
                     CAP_AUDIT_WRITE);
       rc = capng_change_id (uid, gid, CAPNG_DROP_SUPP_GRP);
       if (rc)
diff -ur dbus-1.5.12.orig/dbus/dbus-sysdeps-util-unix.c dbus-1.5.12/dbus/dbus-sysdeps-util-unix.c
--- dbus-1.5.12.orig/dbus/dbus-sysdeps-util-unix.c	2012-03-27 15:48:46.000000000 +0200
+++ dbus-1.5.12/dbus/dbus-sysdeps-util-unix.c	2012-05-27 23:02:39.478187071 +0200
@@ -118,11 +118,15 @@
           dup2 (dev_null_fd, 0);
           dup2 (dev_null_fd, 1);
           
-          s = _dbus_getenv ("DBUS_DEBUG_OUTPUT");
+          /* FIXME - dup2 (dev_null_fd, 2); leads to the problem, that the pid pipe cannot be found:
+	   * most likely the fd of the pidfile gets mangled
+	   *
+	  s = _dbus_getenv ("DBUS_DEBUG_OUTPUT");
           if (s == NULL || *s == '\0')
             dup2 (dev_null_fd, 2);
           else
             _dbus_verbose ("keeping stderr open due to DBUS_DEBUG_OUTPUT\n");
+	  */
         }
 
       if (!keep_umask)
