diff --git a/gnome-session/gsm-consolekit.c b/gnome-session/gsm-consolekit.c
index c20004a..26f0dd5 100644
--- a/gnome-session/gsm-consolekit.c
+++ b/gnome-session/gsm-consolekit.c
@@ -272,6 +272,9 @@ gsm_consolekit_ensure_ck_connection (GsmConsolekit  *manager,
                                              G_CALLBACK (on_active_changed), manager, NULL);
         }
 
+        g_clear_object (&manager->priv->up_client);
+        manager->priv->up_client = up_client_new ();
+
         is_connected = TRUE;
 
  out:
@@ -309,10 +312,8 @@ gsm_consolekit_on_name_owner_changed (DBusGProxy    *bus_proxy,
         }
 
         g_clear_object (&manager->priv->ck_proxy);
-        g_clear_object (&manager->priv->up_client);
 
         gsm_consolekit_ensure_ck_connection (manager, NULL);
-        manager->priv->up_client = up_client_new ();
 
 }
