diff --git a/src/DkThumbsWidgets.cpp b/src/DkThumbsWidgets.cpp
index 3f59c3e..942ac1d 100644
--- a/src/DkThumbsWidgets.cpp
+++ b/src/DkThumbsWidgets.cpp
@@ -1842,7 +1842,6 @@ void DkThumbScrollWidget::addContextMenuActions(const QVector<QAction*>& actions
 		QToolButton* toolButton = new QToolButton(this);
 		toolButton->setObjectName("DkThumbToolButton");
 		toolButton->setMenu(m);
-		toolButton->setAccessibleName(menuTitle);
 		toolButton->setText(menuTitle);
 
 		if (menuTitle == tr("&Sort")) {	// that's an awful hack
