diff --git a/cli/cppcheckexecutor.cpp b/cli/cppcheckexecutor.cpp
index cd57017..9e2e8de 100644
--- a/cli/cppcheckexecutor.cpp
+++ b/cli/cppcheckexecutor.cpp
@@ -33,7 +33,7 @@
 #include <iostream>
 #include <sstream>
 
-#if !defined(NO_UNIX_SIGNAL_HANDLING) && defined(__GNUC__) && !defined(__CYGWIN__) && !defined(__MINGW32__) && !defined(__OS2__)
+#if !defined(NO_UNIX_SIGNAL_HANDLING) && defined(__GLIBC__) && !defined(__CYGWIN__) && !defined(__MINGW32__) && !defined(__OS2__)
 #define USE_UNIX_SIGNAL_HANDLING
 #include <cstdio>
 #include <signal.h>
