--- src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h~	2012-02-11 15:06:02.361171033 -0600
+++ src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h	2012-02-11 15:06:31.718744157 -0600
@@ -52,7 +52,7 @@
 typedef struct _GHashTable GHashTable;
 typedef struct _GInputStream GInputStream;
 typedef struct _GList GList;
-typedef struct _GMutex GMutex;
+typedef union  _GMutex GMutex;
 typedef struct _GPatternSpec GPatternSpec;
 typedef struct _GPollableOutputStream GPollableOutputStream;
 typedef struct _GSocketClient GSocketClient;
