--- old/src/tpfand/settings.py	2009-03-09 09:27:10.000000000 +1100
+++ new/src/tpfand/settings.py	2010-05-05 00:26:48.000000000 +1000
@@ -167,11 +167,11 @@
             self.product_pretty_id = computer.GetProperty('system.hardware.product')
         except:
             print "Warning: unable to get your system model from HAL."
-            self.product_id = None
-            self.product_name = None
-            self.product_pretty_vendor = None
-            self.product_pretty_name = None
-            self.product_pretty_id = None   
+            self.product_id = ''
+            self.product_name = ''
+            self.product_pretty_vendor = ''
+            self.product_pretty_name = ''
+            self.product_pretty_id = ''   
                 
     @dbus.service.method("org.thinkpad.fancontrol.Settings", in_signature='', out_signature='a{is}')
     def get_sensor_names(self):
