--- gtk3-engines-unico-1.0.2.orig/debian/compat
+++ gtk3-engines-unico-1.0.2/debian/compat
@@ -0,0 +1 @@
+7
--- gtk3-engines-unico-1.0.2.orig/debian/rules
+++ gtk3-engines-unico-1.0.2/debian/rules
@@ -0,0 +1,12 @@
+#!/usr/bin/make -f
+# -*- makefile -*-
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+%:
+	dh $@ 
+
+override_dh_install:
+	find debian/gtk3-engines-unico/usr/lib -name \*.la -exec rm {} \;
+	find debian/gtk3-engines-unico/usr/lib -name \*.a -exec rm {} \;
+	dh_install --fail-missing
--- gtk3-engines-unico-1.0.2.orig/debian/copyright
+++ gtk3-engines-unico-1.0.2/debian/copyright
@@ -0,0 +1,20 @@
+Format: http://dep.debian.net/deps/dep5
+Upstream-Name: gtk3-engines-unico
+Upstream-Contact: Andrea Cimitan <andrea.cimitan@canonical.com>
+Source: https://launchpad.net/unico
+
+Files: *
+Copyright: 2011 Canonical Ltd
+License: LGPL-2.1+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+ .
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+ .
+ On Debian systems, the complete text of the GNU Lesser General Public
+ Licence can be found in `/usr/share/common-licenses/LGPL-2.1'
--- gtk3-engines-unico-1.0.2.orig/debian/changelog
+++ gtk3-engines-unico-1.0.2/debian/changelog
@@ -0,0 +1,78 @@
+gtk3-engines-unico (1.0.2-0ubuntu1) precise; urgency=low
+
+  * New upstream release.
+    - Fixed a memory leak
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Thu, 29 Mar 2012 15:01:51 -0400
+
+gtk3-engines-unico (1.0.1-0ubuntu5) precise; urgency=low
+
+  * Backport upstream fixes for i386 look issues and valgrind errors
+
+ -- Sebastien Bacher <seb128@ubuntu.com>  Wed, 29 Feb 2012 21:28:29 +0100
+
+gtk3-engines-unico (1.0.1-0ubuntu4) precise; urgency=low
+
+  * Backport upstream commit "Fixes for inconsistent state"
+
+ -- Sebastien Bacher <seb128@ubuntu.com>  Wed, 22 Feb 2012 21:31:52 +0100
+
+gtk3-engines-unico (1.0.1-0ubuntu3) precise; urgency=low
+
+  * Backported r126, fixes gedit context menu issues (lp: #929138)
+
+ -- Sebastien Bacher <seb128@ubuntu.com>  Thu, 09 Feb 2012 17:18:24 +0100
+
+gtk3-engines-unico (1.0.1-0ubuntu2) precise; urgency=low
+
+  * Backported r125, support correctly the new gtk (lp: #928712)
+  * debian/control: updated gtk requirement
+  * debian/source/format:
+    - switch to format v1, bzr-builddeb doesn't like full source backports
+      using the vcs with v3
+
+ -- Sebastien Bacher <seb128@ubuntu.com>  Wed, 08 Feb 2012 15:16:02 +0100
+
+gtk3-engines-unico (1.0.1-0ubuntu1) oneiric; urgency=low
+
+  * New upstream release.
+    - gtk_render_line doesn't work if y0 == y1 or x1 == x0 (LP: #833958)
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Thu, 01 Sep 2011 15:30:30 -0400
+
+gtk3-engines-unico (1.0-0ubuntu1) oneiric; urgency=low
+
+  * New upstream release.
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Wed, 10 Aug 2011 11:38:14 -0400
+
+gtk3-engines-unico (0.1.0+r74-0ubuntu1) oneiric; urgency=low
+
+  * New upstream release.
+  * debian/control
+    - bump build depends for libgtk-3-dev >= 3.1.6
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Thu, 30 Jun 2011 10:51:38 +0100
+
+gtk3-engines-unico (0.1.0+r69-0ubuntu1) oneiric; urgency=low
+
+  * New upstream snapshot.
+  * debian/control
+    - Updated Vcs-Bzr to point to the proper project
+    - Updated maintainer email to use @ubuntu.com
+    - Updated standards version to 3.9.2
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Fri, 17 Jun 2011 16:18:41 -0400
+
+gtk3-engines-unico (0.1.0+r66-0ubuntu1) oneiric; urgency=low
+
+  * New upstream snapshot.
+    - Fix for a gnome-control-center crasher (LP: #789010)
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Wed, 01 Jun 2011 13:21:40 -0400
+
+gtk3-engines-unico (0.1.0+r64-0ubuntu1) oneiric; urgency=low
+
+  * Initial packaging
+
+ -- Ken VanDine <ken.vandine@canonical.com>  Thu, 19 May 2011 09:46:01 -0400
--- gtk3-engines-unico-1.0.2.orig/debian/watch
+++ gtk3-engines-unico-1.0.2/debian/watch
@@ -0,0 +1,2 @@
+version=3
+https://launchpad.net/unico/+download .*/unico-([0-9.]+)\.tar\.gz
--- gtk3-engines-unico-1.0.2.orig/debian/docs
+++ gtk3-engines-unico-1.0.2/debian/docs
@@ -0,0 +1,2 @@
+NEWS
+README
--- gtk3-engines-unico-1.0.2.orig/debian/control
+++ gtk3-engines-unico-1.0.2/debian/control
@@ -0,0 +1,20 @@
+Source: gtk3-engines-unico
+Priority: optional
+Maintainer: Ken VanDine <ken.vandine@ubuntu.com>
+Build-Depends: debhelper (>= 7.0.50~),
+               autotools-dev,
+               libgtk-3-dev (>= 3.1.14),
+               libglib2.0-dev (>= 2.26.0),
+               libcairo2-dev (>= 1.10) 
+Standards-Version: 3.9.2
+Section: x11
+Homepage: https://launchpad.net/unico
+Vcs-Bzr: https://code.launchpad.net/~ubuntu-desktop/unico/ubuntu
+
+Package: gtk3-engines-unico
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Unico Gtk+ 3 theme engine
+ Unico is a Gtk+ engine that aims to be the more complete yet powerful theming 
+ engine for Gtk+ 3.0 and newer. It’s the first Gtk+ engine written with Gtk+ 
+ style context APIs in mind, using CSS as first class citizen.
--- gtk3-engines-unico-1.0.2.orig/debian/source/format
+++ gtk3-engines-unico-1.0.2/debian/source/format
@@ -0,0 +1 @@
+1.0
