diff -ur glib.orig/glue/list.c glib/glue/list.c
--- glib.orig/glue/list.c	2012-01-25 10:21:25.851685561 +0100
+++ glib/glue/list.c	2012-01-25 10:19:10.473689286 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/glist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_list_get_data (GList *l);
Only in glib/glue: list.c.orig
diff -ur glib.orig/glue/slist.c glib/glue/slist.c
--- glib.orig/glue/slist.c	2012-01-25 10:21:25.851685561 +0100
+++ glib/glue/slist.c	2012-01-25 10:19:51.295688168 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gslist.h>
+#include <glib.h>
 
 /* Forward declarations */
 gpointer gtksharp_slist_get_data (GSList *l);
Only in glib/glue: slist.c.orig
diff -ur glib.orig/glue/thread.c glib/glue/thread.c
--- glib.orig/glue/thread.c	2012-01-25 10:21:25.852685561 +0100
+++ glib/glue/thread.c	2012-01-25 10:20:20.817687342 +0100
@@ -20,7 +20,7 @@
  */
 
 
-#include <glib/gthread.h>
+#include <glib.h>
 
 gboolean glibsharp_g_thread_supported (void);
 
Only in glib/glue: thread.c.orig
