diff -ur objects.orig/custom/shape_typeinfo.c objects/custom/shape_typeinfo.c
--- objects.orig/custom/shape_typeinfo.c	2011-12-26 13:29:15.286204882 +0100
+++ objects/custom/shape_typeinfo.c	2011-12-26 13:29:47.104202723 +0100
@@ -27,8 +27,7 @@
 #include "custom_util.h"
 #include <string.h>
 #include <stdarg.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <libxml/parser.h>
 
 /*

diff -ur tests.orig/test-boundingbox.c tests/test-boundingbox.c
--- tests.orig/test-boundingbox.c	2011-12-26 13:35:49.612177987 +0100
+++ tests/test-boundingbox.c	2011-12-26 13:33:54.293185808 +0100
@@ -29,7 +29,7 @@
 #include <glib-object.h>
 
 #if GLIB_CHECK_VERSION(2,16,0)
-#include <glib/gtestutils.h>
+#include <glib.h>
 #endif
 #include "dialib.h"
 
diff -ur tests.orig/test-objects.c tests/test-objects.c
--- tests.orig/test-objects.c	2011-12-26 13:34:53.584181735 +0100
+++ tests/test-objects.c	2011-12-26 13:36:08.566176610 +0100
@@ -29,7 +29,7 @@
 #include <glib-object.h>
 
 #if GLIB_CHECK_VERSION(2,16,0)
-#include <glib/gtestutils.h>
+#include <glib.h>
 #endif
 
 #include "object.h"
