--- files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js.orig	2012-05-09 16:32:48.828414896 +0200
+++ files/usr/share/cinnamon/applets/power@cinnamon.org/applet.js	2012-05-09 16:24:32.789448815 +0200
@@ -48,7 +48,7 @@
         { name: 'GetPrimaryDevice', inSignature: '', outSignature: '(susdut)' },
         ],
     signals: [
-        { name: 'Changed', inSignature: '' },
+        { name: 'PropertiesChanged', inSignature: 's,a{sv},a[s]' },
         ],
     properties: [
         { name: 'Icon', signature: 's', access: 'read' },
@@ -56,6 +56,20 @@
 };
 let PowerManagerProxy = DBus.makeProxyClass(PowerManagerInterface);
 
+const SettingsManagerInterface = {
+	name: 'org.freedesktop.DBus.Properties',
+	methods: [
+		{ name: 'Get', inSignature: 's,s', outSignature: 'v' },
+		{ name: 'GetAll', inSignature: 's', outSignature: 'a{sv}' },
+		{ name: 'Set', inSignature: 's,s,v', outSignature: '' }
+	],
+	signals: [
+	{name: 'PropertiesChanged', inSignature:'s,a{sv},a[s]', outSignature:''}
+	]
+};
+
+let SettingsManagerProxy = DBus.makeProxyClass(SettingsManagerInterface);
+
 function DeviceItem() {
     this._init.apply(this, arguments);
 }
@@ -131,7 +145,8 @@
             
             this.set_applet_icon_symbolic_name('battery-missing');            
             this._proxy = new PowerManagerProxy(DBus.session, BUS_NAME, OBJECT_PATH);
-
+			this._smProxy = new SettingsManagerProxy(DBus.session, BUS_NAME, OBJECT_PATH);
+            
             let icon = this.actor.get_children()[0];
             this.actor.remove_actor(icon);
             let box = new St.BoxLayout({ name: 'batteryBox' });
@@ -184,7 +199,7 @@
             this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
             this.menu.addSettingsAction(_("Power Settings"), 'gnome-power-panel.desktop');
 
-            this._proxy.connect('Changed', Lang.bind(this, this._devicesChanged));
+            this._smProxy.connect('PropertiesChanged', Lang.bind(this, this._devicesChanged));
             this._devicesChanged();            
         }
         catch (e) {
