diff -u libgnomecups.orig/gnome-cups-i18n.c libgnomecups/gnome-cups-i18n.c
--- libgnomecups.orig/gnome-cups-i18n.c	2011-12-25 17:48:10.398050618 +0100
+++ libgnomecups/gnome-cups-i18n.c	2011-12-25 17:49:23.607045642 +0100
@@ -1,5 +1,5 @@
 #include <config.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include "gnome-cups-i18n.h"
 
 #ifdef ENABLE_NLS
diff -u libgnomecups.orig/gnome-cups-init.h libgnomecups/gnome-cups-init.h
--- libgnomecups.orig/gnome-cups-init.h	2011-12-25 17:48:10.399050618 +0100
+++ libgnomecups/gnome-cups-init.h	2011-12-25 17:49:38.566044582 +0100
@@ -1,8 +1,7 @@
 #ifndef GNOME_CUPS_INIT
 #define GNOME_CUPS_INIT
 
-#include <glib/gtypes.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
