--- refpolicy.orig/policy/mcs	2012-05-29 17:22:20.094453729 +0200
+++ refpolicy/policy/mcs	2012-05-29 17:22:37.327452545 +0200
@@ -117,7 +117,7 @@
 mlsconstrain db_database { drop getattr setattr relabelfrom access install_module load_module get_param set_param }
 	( h1 dom h2 );
 
-mlsconstrain db_schema { drop getattr setattr relabelfrom execute }
+mlsconstrain db_schema { drop getattr setattr relabelfrom }
 	( h1 dom h2 );
 
 mlsconstrain db_table { drop getattr setattr relabelfrom select update insert delete lock }
